example-graphql/src/components/databases/todos.tsx

```diff
@@ -0,0 +1,113 @@
+import { Models } from "appwrite";
+import { useEffect, useRef, useState } from "react";
+import { createDocument, listDocuments, deleteDocument } from "../../appwrite";
+
+const databaseId = import.meta.env.VITE_APPWRITE_DATABASE_ID;
+const collectionId = import.meta.env.VITE_APPWRITE_COLLECTION_ID;
+
+interface Todo extends Models.Document {
+  todo: string;
+  done?: boolean;
+}
+
+export default function Databases() {
+  const [info, setInfo] = useState<
+    { total?: number; todos?: Todo[] } | undefined
+  >();
+
+  const [todo, setTodo] = useState<string>("");
+
+  const effectRan = useRef(false);
+  useEffect(() => {
+    if (effectRan.current === false) {
+      listTodos();
+
+      return () => {
+        effectRan.current = true;
+      };
+    }
+  }, []);
+
+  const listTodos = () => {
+    listDocuments(databaseId, collectionId).then((i) => {
+      setInfo({
+        total: i?.data?.databasesListDocuments?.total,
+        todos: i?.data?.databasesListDocuments?.documents.map((d) =>
+          JSON.parse(d.data)
+        ),
+      });
+    });
+  };
+  const createTodo = async () => {
+    const doc = await createDocument(databaseId, collectionId, {
+      todo,
+    });
+    setTodo("");
+    const parsed =
+      doc?.data?.databasesCreateDocument?.data &&
+      JSON.parse(doc?.data?.databasesCreateDocument?.data);
+    const parsedTodo: Todo = {
+      $id: doc?.data?.databasesCreateDocument._id,
+      ...parsed,
+    };
+    console.log(parsedTodo);
+    setInfo({
+      total: (info?.total || 0) + 1,
+      todos: info?.todos ? [...info.todos, ...[parsedTodo]] : [parsedTodo],
+    });
+  };
+
+  const deleteTodo = async (id: string) => {
+    const doc = await deleteDocument(databaseId, collectionId, id);
+    //TODO: just delete this doc
+    listTodos();
+  };
+
+  // const updateTodo = () => {};
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        gap: "1em",
+      }}
+    >
+      <h1>Todos</h1>
+      <div style={{ display: "flex" }}>
+        <input
+          value={todo}
+          type="text"
+          onChange={(e) => setTodo(e.target.value)}
+        />
+        <button onClick={() => createTodo()}>Add</button>
+      </div>
+      <div className="info">
+        <table>
+          <thead>
+            <tr>
+              <th>Todo</th>
+              <th>Done</th>
+              <th></th>
+            </tr>
+          </thead>
+          <tbody>
+            {info?.total &&
```

Comment:
for some reason, if there are no todos, I get this:

<img width="1164" alt="image" src="https://user-images.githubusercontent.com/1477010/212206778-e2086e9d-95ef-4445-8d30-72d8c1fa2cc7.png">

I think it might need to be:

```suggestion
            {(info?.total || 0) > 0 &&
```

Reply:
fixed